Platforms/Export-Import-Platform.ps1

```diff
@@ -110,136 +129,214 @@ Function Join-ExceptionMessage
+			return
+		}
+	} Else {
+		try {
+			Write-Debug "Setting script to use TLS 1.2"
+			[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12
+		} catch {
+			Write-Error "Could not change SSL settings to use TLS 1.2"
+			Write-Error (Join-ExceptionMessage $_.Exception) -ErrorAction "SilentlyContinue"
+		}
 	}
-	else { return }
-
-    # Create the POST Body for the Logon
-    # ----------------------------------
-    $logonBody = @{ username=$rstusername;password=$rstpassword }
-    $logonBody = $logonBody | ConvertTo-Json
-	try{
-	    # Logon
-	    $logonToken = Invoke-RestMethod -Method Post -Uri $URL_Logon -Body $logonBody -ContentType "application/json"
+
+	# Check that the PVWA URL is OK
+	If ($PVWAURL -ne "") {
+		If ($PVWAURL.Substring($PVWAURL.Length - 1) -eq "/") {
+			$PVWAURL = $PVWAURL.Substring(0, $PVWAURL.Length - 1)
+		}
+	} else {
+		Write-Host -ForegroundColor Red "PVWA URL can not be empty"
+		return
 	}
-	catch
-	{
+
+	Write-Host "Export / Import Platform: Script Started" -ForegroundColor Cyan
+
+	#region [Logon]
+	# Get Credentials to Login
+	# ------------------------
+	$caption = "Export / Import Platform"
+	$msg = "Enter your User name and Password"; 
+	if ($Null -eq $creds) {$creds = $Host.UI.PromptForCredential($caption, $msg, "", "")}
+	if ($null -ne $creds) {
+		$rstusername = $creds.username.Replace('\', '');    
+		$rstpassword = $creds.GetNetworkCredential().password
+
+	} else { return }
+
+	# Create the POST Body for the Logon
+	# ----------------------------------
+	$logonBody = @{ username = $rstusername; password = $rstpassword; concurrentSession = 'true' }
+	$logonBody = $logonBody | ConvertTo-Json
+	try {
+		# Logon
+		$logonToken = Invoke-RestMethod -Method Post -Uri $URL_Logon -Body $logonBody -ContentType "application/json"
+	} catch {
 		Write-Host -ForegroundColor Red $_.Exception.Response.StatusDescription
 		$logonToken = ""
 	}
-    If ($logonToken -eq "")
-    {
-        Write-Host -ForegroundColor Red "Logon Token is Empty - Cannot login"
-        return
-    }
+	If ($logonToken -eq "") {
+		Write-Host -ForegroundColor Red "Logon Token is Empty - Cannot login"
+		return
+	}
 	
-    # Create a Logon Token Header (This will be used through out all the script)
-    # ---------------------------
-    $logonHeader =  New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
-    $logonHeader.Add("Authorization", $logonToken)
-#endregion
+	# Create a Logon Token Header (This will be used through out all the script)
+	# ---------------------------
+	$logonHeader = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+	$logonHeader.Add("Authorization", $logonToken)
+	#endregion
+
+	switch ($PsCmdlet.ParameterSetName) {
+		"Import" {
+			import-platform $PlatformZipPath -error
+		}
+
+		"ImportFile" {
+			foreach ($line in Get-Content $listFile) {
+				if ("" -ne $line) { import-platform $line }
```

Comment:
@bab29 better to use:
[string]::IsNullOrEmpty()
with a -not at the Beginning - this covers all use cases (also space in the line for example)